File: internal/tray/CreateTrayMenu.go
package tray

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/driver/desktop"
	. "github.com/bealearts/kegger/internal/logger"
)

var menu *fyne.Menu

func CreateTrayMenu() {
	//updatesMenu := fyne.NewMenuItem("0 Updates", func() {})

	// menuItems := []*fyne.MenuItem{
	// 	updatesMenu,

	// 	fyne.NewMenuItem("Clean up Celler", func() {
	// 		updatesMenu.ChildMenu.Items = []*fyne.MenuItem{fyne.NewMenuItem("Changed", func() {

	// 		})}
	// 		//menu.Refresh()
	// 	}),
	// 	fyne.NewMenuItemSeparator(),
	// 	fyne.NewMenuItem("About", func() {
	// 		Logger.Info("Hi")
	// 		//dialog.Message("%s", "Do you want to continue?").Title("Are you sure?").Info()
	// 	}),
	// }
	ap := app.New()
	// menu = fyne.NewMenu("Kegger", menuItems...)
	if desk, ok := ap.(desktop.App); ok {
		menu = fyne.NewMenu("Kegger")
		desk.SetSystemTrayMenu((menu))
	}
	Logger.Info("Starting")
	ap.Run()
	//return menu, updatesMenu

	// mUpdates := systray.AddMenuItem("0 Updates Available", "")
	// mUpdates.Disable()

	// mUpdate := systray.AddMenuItem("Update All", "")
	// mUpdates.Disable()
	// go func() {
	// 	<-mUpdate.ClickedCh
	// 	Logger.Info("Update All")
	// 	//brew.ExecCleanup()
	// }()

	// mClean := systray.AddMenuItem("Clean up Celler", "")
	// go func() {
	// 	<-mClean.ClickedCh
	// 	Logger.Info("Clean")
	// 	brew.ExecCleanup()
	// }()

	// systray.AddSeparator()

	// // mPref

	// mAbout := systray.AddMenuItem("About", "")
	// go func() {
	// 	<-mAbout.ClickedCh
	// 	Logger.Info("About")
	// 	dialog.Message("%s", "Kegger - Join the party").Title("About").Info()

	// }()

	// systray.AddSeparator()

	// mQuit := systray.AddMenuItem("Quit", "")
	// go func() {
	// 	<-mQuit.ClickedCh
	// 	Logger.Info("Quit")
	// 	systray.Quit()
	// }()

	// return *mUpdates, *mUpdate
}

File: internal/tray/createTrayMenu.js
import createTrayIcon from "./createTrayIcon.js";
import createDependenciesMenu from "./createDependenciesMenu.js";
import execUpdate from "~/brew/execUpdate.js";
import execCleanup from "~/brew/execCleanup.js";

const icon = await createTrayIcon();

export default function createTrayMenu() {
  return {
    icon: icon,
    tooltip: "Kegger - Join the party",
    items: [
      createDependenciesMenu([]),
      {
        title: "Update All",
        enabled: false,
        onClick: () => execUpdate(),
      },
      {
        title: "Clean up Celler",
        enabled: true,
        onClick: () => execCleanup(),
      },
      {
        title: "<SEPARATOR>",
      },
      {
        title: "Preferences",
        enabled: true,
        items: [
          {
            title: "Start at login",
            enabled: true,
            checked: true,
          },
        ],
      },
      {
        title: "About",
        enabled: true,
      },
      {
        title: "<SEPARATOR>",
      },
      {
        title: "Quit",
        enabled: true,
        onClick: () => Deno.exit(),
      },
    ],
  };
}
